tree_random_test.go

```diff
@@ -148,12 +149,23 @@ func testRandomOperations(t *testing.T, randSeed int64) {
 		if r.Float64() < deleteChance {
 			versions := getMirrorVersions(diskMirrors, memMirrors)
 			if len(versions) > 2 {
-				deleteVersion := int64(versions[r.Intn(len(versions)-1)])
-				t.Logf("Deleting version %v", deleteVersion)
-				err = tree.DeleteVersion(deleteVersion)
-				require.NoError(t, err)
-				delete(diskMirrors, deleteVersion)
-				delete(memMirrors, deleteVersion)
+				i := r.Intn(len(versions) - 1)
+				if i > versions[0] {
```

Comment:
I don't understand this logic. `i` here is the index of a random element in the array, why are we comparing that with a version number?

What we should do here, is to randomly pick whether to delete a single or multiple versions (by `deleteRangeChange`), and if multiple the randomly pick a small range to delete (e.g. 2-5 versions).